### DISCUSSION TITLE
[BUG] command 'github.copilot.generate' not found

### DISCUSSION THREAD

**Discussion by rolimans** [community/community#6634] [Copilot Conversations]
After installing the extension and setting up my Github account in VSCodium I got a message saying the preview wasn't available for my account. Checking the Copilot website it said I was already in the preview.

I uninstalled and re-installed the extension and now when I try to execute any command from the extension I get that the command was not found.

Running on macOS BigSur on a Mac Pro 13 M1

E.g.

`command 'github.copilot.generate' not found`

**Comment by davecheney** [comment]
@rolimans sorry you're having trouble. Can you please try using VSCode and see if that works.

**Reply by khalidnadish** [reply]
i have same problem

**Reply by rolimans** [reply]
VSCode Works ok.

**Reply by shekyu01** [reply]
@rolimans Did you find any workaround?

**Reply by taoliujun** [reply]
same same same

**Reply by davecheney** [reply]
@taoliujun please use vscode, we’ll add support for the other variants in time, but right now the extension will decline to run on them.

**Reply by taoliujun** [reply]
> @taoliujun please use vscode, we’ll add support for the other variants in time, but right now the extension will decline to run on them.

thks, i see and try it when release new.

**Reply by taoliujun** [reply]
> @taoliujun please use vscode, we’ll add support for the other variants in time, but right now the extension will decline to run on them.

It is happened use vscode.

**Reply by davecheney** [reply]
Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by taoliujun** [reply]
> Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

Maybe. I work behind company's network which cross chinese firewall, could you show copilot server domain and i try to exec `ping`.

**Comment by JasonHoku** [comment]
Hey I believe this happened to me but I fixed it by completely stopping and restarting VSCode

**Reply by SI-Abid** [reply]
I have the same problem. I tried re-installing VSCode but the problem remains.

**Comment by Krzysztof-Cieslak** [comment]
Please check if your extension is correctly activated - you should see a small copilot icon in the right bottom corner, on the status bar.

For issues with starting extension on VSCodium/OSS-Code please check - https://github.com/github/feedback/discussions/6629#discussioncomment-1524627

**Reply by s0r3-glitch** [reply]
Hey, I am having this same issue but that comment didnt fix my issue, do you have any other ideas on what I could do to fix this?

**Reply by rolimans** [reply]
Did not fix the issue

**Reply by donhartiggit** [reply]
There is no Github Copilot icon in the status bar - or anywhere else. Followed all the steps, got the verification code, but there is no way to sign in using VS code.

**Comment by alecavallo** [comment]
I have exactly the same problem in vscode on macOS big sur.
I tried uninstalling GitHub copilot, removing the folder _./.vscode/extensions/GitHub.copilot-1.5.3432_ and installing the extension again. But the problem persists.
I installed vscode on a Linux machine and GitHub Copilot works without issues.

**Reply by DonRichards** [reply]
Do you have any other plugins that suggest code like "Kite" or something? This might be causing some sort of conflict. Disabling Kite seems to have corrected my issue I think. I'm looking through the plugins and disabling everything I think might be adding code suggestions so I can say with confidence that Copilot is working.

**Reply by alecavallo** [reply]
I had installed tabnine but now it's deactivated.
I think the issue is caused by company's firewall blocking the requests to copilot's servers. Do hoy know its names or IP so I can check if I'm able to connect to it or, in case it's blocking the connection, to request the whitelisting of the servers?

**Reply by ajhawkings** [reply]
Am having the same problem, specifically only on a work machine where vscode is installed as a zip (I initially thought the installation was the problem), but now I am assuming it is a firewall problem... Disappointing that there is no reference to copilot's IPs or domains to unblock anywhere :/

**Comment by rolimans** [comment]
It seems to be a problem related on how VSCodium authenticates (using PAT's). When using the extension with version 1.1.1909 it works fine. I suspect it is something related with the copilot telemetry agreement that never shows up on VSCodium.

**Reply by PatoLocos** [reply]
After a whole week trying to find a solution, this worked for me using Windows 10 latest version (as of 28/10/2021) and latest version of visual studio code.

**Comment by rolimans** [comment]
When using version 1.1.1909 I get the following output
```
[INFO] [auth] [2021-10-26T21:22:42.982Z] Invalid copilot token: missing token
[ERROR] [ghostText] [2021-10-26T21:22:42.983Z] Error on ghost text request Error: User not authorized
```

**Reply by davecheney** [reply]
@rolimans please see #6637

**Reply by rolimans** [reply]
I've imagined it has something to do with using an older version, but newer versions don't work, as the issue suggests

**Comment by alexandreleduc** [comment]
Re-authenticating with GitHub doesn't fix it for me, however when I disconnect from my work VPN, copilot works just fine. The other extension I have that uses the GitHub account (GitLens) seems to work fine on my work VPN.  Note that I'm having the _command 'github.copilot.generate' not found_ error in VS Code

**Reply by PROxZIMA** [reply]
I'm on VSCodium `1.62.0` with GitHub Copilot `1.7.3689` and no other clutter. Still I'm getting the `command 'github.copilot.generate' not found` error :(

**Reply by Tumi-D** [reply]
Facing the same issue non of the suggested solutions are  working

**Comment by KSH-code** [comment]
any of MacOS user should check my answer https://github.com/github/feedback/discussions/7178#discussioncomment-1604641 
clean entire vscode files and then install.

**Comment by lmagnien** [comment]
Maybe the error is related to the use of a proxy.

I installed copilot 1.7 on vs code 1.62.0.
When I used copilot behind a proxy I got the same error : command 'github.copilot.generate' not found.
I tried without proxy and I received a notification to accept the terms of use. I accepted and then it worked. 
I switch again behind the proxy and now, the error disappear but copilot still does not work. 

Searching in the discussions, I saw that proxy would not be supported yet :

https://github.com/github/feedback/discussions/6649#discussioncomment-1525131

**Reply by davecheney** [reply]
that is correct, proxies are currently not supported.

**Reply by lmagnien** [reply]
Thank you for your answer. Any idea when it will be supported?

**Reply by taoliujun** [reply]
VPN is used by lot of company because of chinese firewall. -_-

**Reply by volreis** [reply]
workaround: use some proxy software (ex. proxifier, proxychains) to proxy the vscode connection

**Reply by herrsergio** [reply]
I have the same issue, company is using Zscaler and Copilot is not working.

**Reply by mirmostafa** [reply]
It seems that this is happening for many developers including me. I cannot solve it yet 😕.

**Reply by taoliujun** [reply]
> workaround: use some proxy software (ex. proxifier, proxychains) to proxy the vscode connection

cannot solve it after set http_proxy.

**Comment by fubits1** [comment]
still happening 
![image](https://github.com/community/community/assets/13749354/f74befa2-0fbf-434b-bbe4-ec858619e4e0)

**Reply by rolimans** [reply]
Yep, it was fixed for a long time, but now the problem seems to be back. Not that big deal now tho, because GitHub Copilot Chat (Cmd+I) is able to generate code.

**Reply by abhiparate14** [reply]
but what if want those 10 suggestions in the side panel.
`CTRL + ENTER`
gives the same error

**Reply by Matej4386** [reply]
Same here, everything is working except CTRL+Enter - gives the error.

**Reply by Pl8tinium** [reply]
yesssssss i neeeeeed the suggestions panel

**Reply by Pl8tinium** [reply]
this thread about to blow up

**Reply by Matej4386** [reply]
I have downgraded GitHub Copilot to v1.174.0 and now it is working.

**Reply by Micky0256** [reply]
Downgrading the git copilot extension is working for me too

**Reply by rioleex** [reply]
Yeah，I install an older version (1.174.0) , now it's working! thank you.

**Reply by houchaowei** [reply]
升级一下 vscode 的 copilot 的插件版本就可以了~

**Comment by babymonie** [comment]
Having the same issue now, on the latest version, I'm on windows

I've tried removing the caches as some people mentioned yet it still happens

**Reply by GatePara** [reply]
me too

**Reply by GatePara** [reply]
v1.175.0 working well

**Comment by hhddd123** [comment]
on the latest version,It's back.

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
GitHub Copilot
v1.176.0
![GitHub Copilot
v1.176.0](https://github.com/community/community/assets/46166740/f1869ac1-e796-44f4-95ff-a0dda19f8d30)

[Window Title]
Visual Studio Code

[Main Instruction]
Der Befehl "GitHub Copilot: Open Completions Panel" hat zu einem Fehler geführt.

[Content]
command 'github.copilot.generate' not found

[OK]
![github.copilot.generate](https://github.com/community/community/assets/46166740/0f8b7094-a692-4eed-a6a7-1b2d05074d6c)

**Comment by n-r-w** [comment]
GitHub Copilot v1.176.0
![image](https://github.com/community/community/assets/38435980/d51bdbaa-be0e-46af-9ade-dc0acf94412a)

v1.175.0 working fine

**Reply by lantrix** [reply]
Confirmed also downgrade to 1.175.0 fixed this

**Reply by mikprin** [reply]
How to downgrade to lower version of extension?
UPD: found here https://code.visualstudio.com/updates/v1_30#_install-previous-versions

**Reply by NeoMetric** [reply]
yes, downgrading to 1.175.0 fixed the issue for me. Thanks for the suggestion.

**Comment by GurramManojReddy** [comment]
I searched everywhere to find the solutions. I tried deleting all the extensions and all the extensions cache. The best solution I can find is by downgrading to the previous versions, like 1.166.0. Newer versions are having the issues.
You can downgrade by right clicking the extension and choosing the version. Then don't forget to reload the extension.

**Comment by guusw** [comment]
`command 'github.copilot.generate' not found`
Both from command menu and shortcut, doesn't seem to work anymore, downgraded to `v1.175.0` it still works in that version.
In  the latest release version (and pre-release) it doesn't work anymore

**Comment by n-r-w** [comment]
v1.177.0 working without issue
